package.json

```diff
@@ -87,31 +35,117 @@
+          "type": "string",
+          "default": "",
+          "order": 2
+        }
       }
     },
-    "rulesToDisableWhileFixing": {
-      "title": "Disable specific rules from fixes",
-      "description": "Prevent rules from being auto-fixed by ESLint. Applies to fixes made during saves as well as when running the `Linter Eslint: Fix File` command.",
-      "type": "array",
-      "default": [],
-      "items": {
-        "type": "string"
+    "disabling": {
+      "type": "object",
+      "order": 6,
+      "properties": {
+        "disableWhenNoEslintConfig": {
+          "title": "Disable when no ESLint config is found (in package.json or .eslintrc)",
+          "type": "boolean",
+          "default": true,
+          "order": 1
+        },
+        "ignoreFixableRulesWhileTyping": {
+          "title": "Ignore fixable rules while typing",
+          "description": "Have the linter ignore all fixable rules during linting when editing a document. The list is automatically updated on each lint job, and requires at least one run to be populated. Only supported when using ESLint v4+.",
+          "type": "boolean",
+          "default": false,
+          "order": 2
+        },
+        "rulesToDisableWhileFixing": {
+          "title": "Disable specific rules from fixes",
+          "description": "Prevent rules from being auto-fixed by ESLint. Applies to fixes made during saves as well as when running the `Linter Eslint: Fix File` command.",
+          "type": "array",
+          "default": [],
+          "items": {
+            "type": "string"
+          },
+          "order": 3
+        },
+        "rulesToSilenceWhileTyping": {
+          "title": "Silence specific rules while typing",
+          "description": "Useful when Atom fixes errors on save like `no-trailing-spaces` or `eol-last`.",
+          "type": "array",
+          "default": [],
+          "items": {
+            "type": "string"
+          },
+          "order": 4
+        }
       }
     },
-    "ignoreFixableRulesWhileTyping": {
-      "title": "Ignore fixable rules while typing",
-      "description": "Have the linter ignore all fixable rules during linting when editing a document. The list is automatically updated on each lint job, and requires at least one run to be populated. Only supported when using ESLint v4+.",
-      "type": "boolean",
-      "default": false
+    "advanced": {
+      "type": "object",
+      "order": 7,
+      "properties": {
+        "disableEslintIgnore": {
+          "title": "Don't use .eslintignore files",
+          "type": "boolean",
+          "default": false,
+          "order": 1
+        },
+        "disableFSCache": {
+          "title": "Disable FileSystem Cache",
+          "description": "Paths of node_modules, .eslintignore and others are cached",
+          "type": "boolean",
+          "default": false,
+          "order": 2
+        },
+        "showRuleIdInMessage": {
```

Comment:
One could make the argument this should go in the "general uncategorized" area, but I think it's fine leaving this down here.

Reply:
I don't really understand why someone would want to turn that off.  Do you know off-hand the reason for the option to exist at all?

Reply:
It would make the output cleaner and easier to read at the cost of being much harder to look up rules in the docs or write inline disable rules. _I_ would never turn this off, but I can see the motivation. I think it's reasonable to argue for removing it entirely.

Reply:
Looks like it was added in https://github.com/AtomLinter/linter-eslint/commit/72cc3cb24be8e488030bb9d8eccae1cec0f7355c, I was hoping for a bit more context as to _why_, but no luck.

Reply:
If I remember correctly he wanted to remove it completely when doing the rewrite, but I made him keep it around (this was before the rule ID even linked to the relevant page, so it was very helpful for googleing an error). The compromise was an option that defaulted enabled.

Reply:
When I said you could argue to remove it, I meant you could remove the setting. Removing the rule id for everybody would be incredibly mean.

Reply:
Oh personally I would be for removing the setting, but there are people out there that don't like the "noise".